File: services/api-gateway/app/schemas/task_intent.py
"""Task intent schema — mirrors packages/types TaskIntent."""

from typing import Any, Literal

from pydantic import BaseModel, Field

TaskIntentPriority = Literal["low", "normal", "high", "critical"]


class TaskIntentSchema(BaseModel):
    """Declared intent behind a user task."""

    kind: str = Field(..., description="Intent category")
    description: str = Field(..., description="Natural-language or structured description")
    parameters: dict[str, Any] | None = Field(default=None)
    priority: TaskIntentPriority | None = Field(default=None)

File: packages/types/src/contracts/task-intent.ts
/**
 * Declared intent behind a user task — what the user wants accomplished.
 */
export interface TaskIntent {
  /** Intent category (e.g. "automate", "research", "draft"). */
  readonly kind: string;
  /** Natural-language or structured intent payload. */
  readonly description: string;
  /** Optional structured parameters for planners (Hermes, Phase 3+). */
  readonly parameters?: Readonly<Record<string, unknown>>;
  /** Relative urgency; orchestrator may reorder queue (Phase 3+). */
  readonly priority?: TaskIntentPriority;
}

/** Task intent priority levels. */
export type TaskIntentPriority = "low" | "normal" | "high" | "critical";
